components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java

```diff
@@ -258,7 +260,8 @@ protected boolean handlePostAuthentication(Event event, String userName, UserSto
             if (NumberUtils.isNumber(userClaimValue)) {
                 unlockTime = Long.parseLong(userClaimValue);
             }
-            if (unlockTime != 0 && System.currentTimeMillis() >= unlockTime || currentFailedAttempts > 0) {
+            if (unlockTime != 0 && System.currentTimeMillis() >= unlockTime || currentFailedAttempts > 0
```

Comment:
Can you please explain the logic here. if the isAccountLockBypassRoleExist(userStoreManager, userName) it always updates the user claims which is wrong.

Reply:
If the login attempt is successful we need to clear locked claims right? Lets say admit locks an account then the bypass role attaches on to the user. In that case we anyway bypass locking feature but if the login attempt is successful we need to clear claim values. Otherwise claim shows as user is locked but get the access.

Reply:
This is not the correct place to do what you expect. If you do here, it will do a setUserClaimValue database call in everytime that kind of user is login. 

You can do it in a separate if condition...

if(accountLockClaim=true && isAccountLockBypassRoleExist(userStoreManager, userName) ) {
resetClaims;
}

Reply:
Done